torch/csrc/jit/script/compiler.cpp

```diff
@@ -1194,14 +1194,25 @@ struct to_ir {
     // ordered set, because we want deterministic graph output
     std::set<std::string> mutated_variables;
 
+    // var is only defined in one branch save error in case it's used later
+    auto err = [&](const std::string& v, const std::string& branch) {
+      ErrorReport error(stmt);
```

Comment:
this will get thrown when we try to do `getVar` to unify value types?

Reply:
No, this uses the same mechanism we use when we cannot unify the type of two values. We don't emit an error now, and cache the error msg if they try to use the var again later.  

https://github.com/pytorch/pytorch/blob/4d676d53a67e610689afbb34deabcac6ba8b8c50/torch/csrc/jit/script/compiler.cpp#L429